omen2/omen.py

```diff
@@ -205,3 +206,9 @@ def _create_if_needed(self):
     def schema(cls, version):
         """Override this to return a schema for a given version."""
         ...
+
+    @contextmanager
+    def transaction(self):
+        for tab in self.tables:
```

Comment:
with ExitStack() as stack:
    for mgr in ctx_managers:
        stack.enter_context(mgr)